src/ucp/core/ucp_request.h

```diff
@@ -167,7 +167,8 @@ struct ucp_request {
 
                 struct {
                     uintptr_t         remote_request; /* pointer to the send request on receiver side */
-                    ucp_request_t     *rreq;
+                    ucp_request_t     *rreq;          /* pointer to the receive request */
+                    size_t            length;         /* the length of the data that could be sent */
```

Comment:
.. that should be fetched from sender side */

Reply:
👍 
done